vanetza/asn1/fetch_iso_asn1_files.cmake

```diff
@@ -18,4 +18,36 @@ file(READ "${DIRECTORY}/ISO19091.asn" _content)
 string(REPLACE "HeadingConfidence" "HeadingConfidenceIso" _content "${_content}")
 string(REPLACE "Heading" "HeadingIso" _content "${_content}")
 string(REPLACE "SpeedConfidence" "SpeedConfidenceIso" _content "${_content}")
-file(WRITE "${DIRECTORY}/ISO19091.asn" "${_content}")
\ No newline at end of file
+file(WRITE "${DIRECTORY}/ISO19091.asn" "${_content}")
+
+file(DOWNLOAD
+    https://standards.iso.org/iso/ts/19321/ed-2/en/ISO19321IVIv2.asn
+    "${DIRECTORY}/ISO19321.asn")
+file(READ "${DIRECTORY}/ISO19321.asn" _content)
+string(REPLACE "[[" "" _content "${_content}")
+string(REPLACE "]]" "" _content "${_content}")
+string(REPLACE "DriverCharacteristics" "DriverCharacteristicsIso" _content "${_content}")
+string(REPLACE "TrailerCharacteristics" "TrailerCharacteristicsIso" _content "${_content}")
+string(REPLACE "Temperature" "TemperatureIso" _content "${_content}")
+file(WRITE "${DIRECTORY}/ISO19321.asn" "${_content}")
+
+file(DOWNLOAD
+    "https://forge.etsi.org/rep/ITS/asn1/is_ts103301/-/raw/v2.1.1/iso-patched/ISO14906(2018)EfcDsrcGenericv7-patched.asn"
+    "${DIRECTORY}/ISO14906-1-7.asn")
+
+file(DOWNLOAD
+    "https://forge.etsi.org/rep/ITS/asn1/is_ts103301/-/raw/v2.1.1/iso-patched/ISO14906(2018)EfcDsrcApplicationv6-patched.asn"
+    "${DIRECTORY}/ISO14906-0-6.asn")
+file(READ "${DIRECTORY}/ISO14906-0-6.asn" _content)
+string(REPLACE "StationType" "StationTypeIso" _content "${_content}")
+file(WRITE "${DIRECTORY}/ISO14906-0-6.asn" "${_content}")
+
+file(DOWNLOAD
+    "https://forge.etsi.org/rep/ITS/asn1/is_ts103301/-/raw/v2.1.1/iso-patched/ISO14823-missing.asn"
+    "${DIRECTORY}/ISO14823.asn")
+
+file(DOWNLOAD
+    "https://forge.etsi.org/rep/LIBS/LibIts/-/raw/798bfce82f199cdfdef4289098bf900832e3932c/asn1/IS/IVI_ISO_TS_17419.asn"
```

Comment:
Would it be cleaner to use https://standards.iso.org/iso/ts/17419/TS%2017419%20ASN.1%20repository/TS17419_2014_CITSapplMgmtIDs.asn here instead or is it good enough like this? I did not find any newer edition.

Reply:
I am fine with that URL. The tricky thing is just to have a compatible set of ASN.1 files…